### PR TITLE
Removing cffi, ctypes, and pypy support

### PR DESCRIPTION
**Pull request by mayantaylor**
Removing build support and corresponding documentation for cffi, ctypes, and the PyPy Python implementation. This PR also drops support for Python 2.

### CODE REVIEW

**Review by ritvikrao — DISMISSED** [review]
Looks good to me on my own machine (apple intel) and Delta. I'll approve it, but others should check the build as well

File: charm4py/__init__.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import sys
-if sys.version_info < (2, 7, 0):
-    raise RuntimeError('Charm4py requires Python 2.7 or higher')
+if sys.version_info < (3, 0, 0):
+    raise RuntimeError('Charm4py requires Python 3.0 or higher')
```
Comment (matthiasdiener) [inline charm4py/__init__.py:3]: maybe:

```suggestion
if sys.version_info < (3, 7, 0):
    raise RuntimeError('Charm4py requires Python 3.7 or higher')
```

**Review by matthiasdiener — DISMISSED** [review]

File: charm4py/charm.py [hunk 2]
```diff
@@ -541,18 +541,14 @@ def registerMainModule(self):
...
                        py_impl + "). Using '" +
                        self.lib.name + "' interface to access Charm++")
             if py_impl == 'PyPy':
-                if self.lib.name != 'cffi':
-                    out_msg += ", **WARNING**: cffi recommended for best performance"
-            elif self.lib.name != 'cython':
-                out_msg += ", **WARNING**: cython recommended for best performance"
-            print(out_msg)
+                raise Charm4PyError('PyPy is no longer supported. Use CPython instead')
             if sys.version_info < (3,0,0):
```
Comment (ritvikrao) [inline charm4py/charm.py:550]: change this to 3.7.0 as well (and the comment in the error as well)

File: charm4py/charm.py [hunk 3]
```diff
@@ -541,18 +541,14 @@ def registerMainModule(self):
             import platform
             from . import charm4py_version
             py_impl = platform.python_implementation()
-            out_msg = ("Charm4py> Running Charm4py version " + charm4py_version +
+            print("Charm4py> Running Charm4py version " + charm4py_version +
                        " on Python " + str(platform.python_version()) + " (" +
                        py_impl + "). Using '" +
                        self.lib.name + "' interface to access Charm++")
             if py_impl == 'PyPy':
```
Comment (ritvikrao) [inline charm4py/charm.py:548]: I think you should change this line to be:
```suggestion
            if py_impl != 'Cython':
```

so it covers all possible Python implementations.

**Review by ritvikrao — APPROVED** [review]

**Pull request merged**